src/java/com/unifina/signalpath/ModuleOption.java

```diff
@@ -125,7 +123,9 @@ public static boolean validate(Object option) {
 				List<Map<String, String>> possibleValues = (List<Map<String, String>>) opt.get("possibleValues");
 				boolean found = false;
 				for (Map<String, String> value : possibleValues) {
-					if (value.get("value") == opt.get("value") || value.get("value").equals(opt.get("value"))) {
+					final String val = value.get("value");
+					final Object optValue = opt.get("value");
+					if (val.equals(optValue)) {
```

Comment:
Can we add `val != null && ` infront?

Reply:
semantics are changed by this patch. Definitely add a null check.

Notice though that `opt.get("value")` is unchanged inside the loop, so it should be taken outside. That whole inner loop thing should probably be something like
```
String selected = (String)opt.get("value");
return possibleValues.stream().anyMatch(v -> ((String)v.get("value")).equals(selected));
```